src/libmongoc/tests/unified/runner.c

```diff
@@ -845,11 +844,16 @@ test_setup_initial_data (test_t *test, bson_error_t *error)
       /* Check if the server supports majority write concern on 'drop' and
        * 'create'. */
       if (semver_cmp_str (&test_runner->server_version, "3.4") >= 0) {
-         drop_opts = bson_new ();
          mongoc_write_concern_append (wc, drop_opts);
-         create_opts = bson_new ();
          mongoc_write_concern_append (wc, create_opts);
       }
+
+      if (is_topology_type_sharded (test_runner->topology_type)) {
+         // From spec: "test runner SHOULD use a single mongos for handling initialData"
+         BSON_APPEND_INT32 (drop_opts, "serverId", 1);
+         BSON_APPEND_INT32 (create_opts, "serverId", 1);
```

Comment:
Curious: why a hardcoded serverId vs. pinning via a client session?

Reply:
I expect either works. Session pinning only occurs on mongos when [using a transaction](https://github.com/mongodb/specifications/blob/ce9b56ba3ed76cb4ebca9d4b760260548007dce6/source/transactions/transactions.md#mongos-pinning) or [using a load balancer](https://github.com/mongodb/specifications/blob/ce9b56ba3ed76cb4ebca9d4b760260548007dce6/source/load-balancers/load-balancers.md#connection-pinning). Keeping the hard coding for simplicity.